db/schema_tables.cc

```diff
@@ -799,11 +799,12 @@ future<table_schema_version> calculate_schema_digest(distributed<service::storag
         auto rs = co_await db::system_keyspace::query_mutations(db, NAME, table);
         auto s = db.local().find_schema(NAME, table);
         for (auto&& p : rs->partitions()) {
-            auto mut = co_await unfreeze_gently(p.mut(), s);
-            auto partition_key = value_cast<sstring>(utf8_type->deserialize(mut.key().get_component(*s, 0)));
+            auto dk = dht::decorate_key(*s, p.mut().key());
+            auto partition_key = value_cast<sstring>(utf8_type->deserialize(dk.key().get_component(*s, 0)));
```

Comment:
Looks like you don't need to decorate the key.

Reply:
done in v3